src/ui/reducers/reactDevTools.ts

```diff
@@ -2,27 +2,54 @@ import { compareNumericStrings } from "protocol/utils";
 import { UIState } from "ui/state";
 import { ReactDevToolsState } from "ui/state/reactDevTools";
 import { ReactDevToolsAction } from "ui/actions/reactDevTools";
+import { ExecutionPoint } from "@recordreplay/protocol";
 
 export default function update(
-  state = { annotations: [], currentPoint: null, hasReactComponents: false },
+  state = {
+    annotations: [],
+    currentPoint: null,
+    hasReactComponents: false,
+    firstOpAnnotationPoint: null,
+    lastProtocolCheckFailed: false,
+  },
   action: ReactDevToolsAction
 ): ReactDevToolsState {
   switch (action.type) {
     case "add_annotations": {
       const annotations = [...state.annotations, ...action.annotations];
       annotations.sort((a1, a2) => compareNumericStrings(a1.point, a2.point));
-      const hasReactComponents = annotations.some(
-        annotation => annotation.message.event === "operations"
-      );
-      return { ...state, annotations, hasReactComponents };
+
+      let firstOpAnnotationPoint: ExecutionPoint | null = null;
+      {
+        for (const { message, point } of annotations) {
+          if (message.event === "operations") {
+            firstOpAnnotationPoint = point;
+            break;
+          }
+        }
+      }
+      const hasReactComponents = firstOpAnnotationPoint !== null;
```

Comment:
couple of stylistic points:

1. lets rename `firstOpAnnotationPoint` to `initPoint` or something more intent based and less implementation based.
2. I think `_.find` would be better for finding the first operation.

```js
const firstOperations = find(annotations, annotation => annotation.message.event == "operations")
const initPoint = firstOperations?.point
const hasReactComponents = !!initPoint
...
```